common/kokkos-sampler/kp_sampler_skip.cpp

```diff
@@ -1,350 +1,420 @@
+          (finalizeFunction)dlsym(childLibrary, "kokkosp_finalize_library");
+
+      if (NULL != initProfileLibrary) {
+        (*initProfileLibrary)(loadSeq + 1, interfaceVer, devInfoCount,
+                              deviceInfo);
+      }
+
+      if (tool_verbosity > 0) {
+        std::cout << "KokkosP: Function Status:\n";
+        std::cout << "KokkosP: begin-parallel-for:      "
+                  << ((beginForCallee == NULL) ? "no" : "yes") << "\n";
+        std::cout << "KokkosP: begin-parallel-scan:     "
+                  << ((beginScanCallee == NULL) ? "no" : "yes") << "\n";
+        std::cout << "KokkosP: begin-parallel-reduce:   "
+                  << ((beginReduceCallee == NULL) ? "no" : "yes") << "\n";
+        std::cout << "KokkosP: end-parallel-for:        "
+                  << ((endForCallee == NULL) ? "no" : "yes") << "\n";
+        std::cout << "KokkosP: end-parallel-scan:       "
+                  << ((endScanCallee == NULL) ? "no" : "yes") << "\n";
+        std::cout << "KokkosP: end-parallel-reduce:     "
+                  << ((endReduceCallee == NULL) ? "no" : "yes") << "\n";
+      }
+    }
+  }
+
+  free(envBuffer);
+
+  uniqID = 1;
+
+  if (0 > tool_seed) {
+    srand(time(NULL));
+    if (tool_verbosity > 0) {
+      std::cout << "KokkosP: Seeding random number generator using clock for "
+                   "random sampling.\n";
+    }
+  } else {
+    srand(tool_seed);
+    if (tool_verbosity > 0) {
+      std::cout << "KokkosP: Seeding random number generator using seed "
+                << tool_seed << " for random sampling.\n";
+    }
+  }
+
+  const char* tool_probability = getenv("KOKKOS_TOOLS_SAMPLER_PROB");
+
+  if (NULL != tool_probability) {
+    //  Read sampling probability as a float between 0 and 100, representing
+    //  a percentage that data should be gathered.
+    //  Utility reasons about probability as a double between 0.0 and 1.0.
+    tool_prob_num = atof(tool_probability);
+    if (tool_prob_num > 100.0) {
+      std::cout << "KokkosP: The sampling probability value is set to be "
+                   "greater than 100.0. The probability for the sampler will "
+                   "be set to 100 percent; all of the invocations of a Kokkos "
+                   "kernel will be profiled.\n";
+      tool_prob_num = 100.0;
+    } else if (tool_prob_num < 0.0) {
+      std::cout
+          << "KokkosP: The sampling probability value is set to be a negative "
+             "number. The sampler's probability will be set to 0 percent; none "
+             "of the invocations of a Kokkos kernel will be profiled.\n";
+      tool_prob_num = 0.0;
+    }
+    if (tool_verbosity > 0) {
+      std::cout << "KokkosP: Probability for the sampler set to: "
+                << tool_prob_num << "\n";
+    }
+    kernelSampleSkip = 1;
+    return;
+  }
+
+  const char* tool_sample = getenv("KOKKOS_TOOLS_SAMPLER_SKIP");
+  if (NULL != tool_sample) {
+    tool_prob_num    = 100.0;
```

Comment:
this means SamplerSkip is prioritized over random setting, because it overwrites the previous setting

Reply:
Yes, this is fixed now.